akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -245,7 +248,8 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
                 masterTerminator.registerConnection(connectionTerminator)(fm.executionContext)
                 future // drop the terminator matValue, we already registered is which is all we need to do here
             }
-
+            // attribute is also added to serverLayer but we also want it to cover the full stack
+            .addAttributes(cancellationStrategyAttributeForDelay(settings.streamCancellationDelay))
```

Comment:
Hmm, do we really, what does it help with in the context of the stream of incoming connections, there's no bidi-stuff going on here?

Reply:
I think it's the per-connection stream here, isn't it? I thought, `fullLayer` above would add TCP and TLS which are bidi-like, so adding the streams cancellation delay here as well could be interesting.

Reply:
Yep, NVM, I thought it was on the outer layer of stream but it isn't.